graph/api/src/main/java/org/jboss/windup/graph/model/comparator/FilePathComparator.java

```diff
@@ -30,39 +32,47 @@
     @Override
     public int compare(String o1, String o2)
     {
-        // if they are exactly the same, just short circuit everything
-        // and return 0
-        if (o1.equals(o2))
+        ExecutionStatistics.get().begin("FilePathComparator");
+        try
         {
-            return 0;
-        }
+            // if they are exactly the same, just short circuit everything
+            // and return 0
+            if (o1.equals(o2))
+            {
+                return 0;
+            }
 
-        // split by the path separator (/ or \)
-        Path path1 = Paths.get(o1);
-        Path path2 = Paths.get(o2);
+            // split by the path separator (/ or \)
+            Path path1 = Paths.get(o1);
+            Path path2 = Paths.get(o2);
 
-        if (path1.getNameCount() != path2.getNameCount())
-        {
-            // if there are differing number of path elements, compare based on number of segments
-            return path1.getNameCount() - path2.getNameCount();
-        }
-        else
-        {
-            // otherwise, compare each segment
-            for (int i = 0; i < path1.getNameCount(); i++)
+            if (path1.getNameCount() != path2.getNameCount())
```

Comment:
I think that it looks better aesthetically, as it groups things at the same level closer together. I don't know if I have ever seen a path sort that didn't do this at some level, since the underlying data is really a tree.